feat: OPTIC-1746: Improve global error message handling by showing toast messages in favour of modals with increased Sentry reporting

bmartel opened the pull request:
<!-- Follow Merge -->
This PR was automatically generated via Follow Merge. [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/13783174190)
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.
- 
<!-- Follow Merge -->


<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->
This pull request introduces several changes to streamline error handling and improve the user experience by integrating toast notifications across various components as a replacement of a modal message. There are also improvements to Sentry error tracking of non actionable errors which will no longer be shown to the user through updates to the API provider to handle global error messages more effectively.

### UI Enhancements:
* Added `ToastProvider` and `ToastViewport` components to `App.jsx`, `Form.jsx`, and `Modal.jsx` to enable toast notifications which provide a less intrusive way of communicating failures over the previous default of a Modal message. [[1]](diffhunk://#diff-36c1030ca59fcd90ceb45a672287abc0f8a01670167dc4d269eb7f44425b85b0R61-L65) [[2]](diffhunk://#diff-5e0cbad5e9c63daf36c8671ecd3969166f5812b795aac37834c1b1cedf90b90bR21) [[3]](diffhunk://#diff-5e0cbad5e9c63daf36c8671ecd3969166f5812b795aac37834c1b1cedf90b90bR69) [[4]](diffhunk://#diff-5e0cbad5e9c63daf36c8671ecd3969166f5812b795aac37834c1b1cedf90b90bR91) [[5]](diffhunk://#diff-83ca9f15e39cf9246a473dabea8f46211cb3770f0f37a9bbf9e0026a3b80c358R11) [[6]](diffhunk://#diff-83ca9f15e39cf9246a473dabea8f46211cb3770f0f37a9bbf9e0026a3b80c358L30-R36) [[7]](diffhunk://#diff-83ca9f15e39cf9246a473dabea8f46211cb3770f0f37a9bbf9e0026a3b80c358R49)

### Error Handling Improvements:
* Enhanced Sentry integration by adding environment-specific configurations and new methods `captureMessage` and `captureException` for better error tracking.
* Updated `ApiProvider` to support improved global error messages, including the use of toast notifications for 4xx errors and enhanced error logging to Sentry for non-4xx errors. [[1]](diffhunk://#diff-a19bbea04128de7fc2af68ba09aedf771c93e474e5b5323e3defa3c77ef9660eR16-R23) [[2]](diffhunk://#diff-a19bbea04128de7fc2af68ba09aedf771c93e474e5b5323e3defa3c77ef9660eR37-R54) [[3]](diffhunk://#diff-a19bbea04128de7fc2af68ba09aedf771c93e474e5b5323e3defa3c77ef9660eL68-L77) [[4]](diffhunk://#diff-a19bbea04128de7fc2af68ba09aedf771c93e474e5b5323e3defa3c77ef9660eL88-R221) [[5]](diffhunk://#diff-a19bbea04128de7fc2af68ba09aedf771c93e474e5b5323e3defa3c77ef9660eL143-R241)

### Testing
* TBD

### Feature Flags:
* Introduced a new feature flag `fflag_feat_front_optic_1746_improve_global_error_messages_short` to control the rollout of these changes.

bmartel committed:
feat: OPTIC-1746: Improve global error message handling by showing toast messages in favour of modals and increased Sentry reporting

bmartel committed:
linting

bmartel committed:
Capture error boundary in sentry, remove stacktrace from Modal

bmartel committed:
linting

bmartel committed:
don't throw an error in test from useToast as it forces every consumer of the hook to mock it in order to stop it from causing errors even if unused

bmartel commented on web/apps/labelstudio/src/components/Modal/Modal.jsx:
```diff
@@ -40,6 +46,7 @@ const standaloneModal = (props) => {
           }}
           animateAppearance={animate}
         />
+        <ToastViewport />
```
```suggestion
        {!props.simple && <ToastViewport />}
```

bmartel committed:
Update web/apps/labelstudio/src/components/Modal/Modal.jsx

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13697181579)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1746'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13697181579

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13697181579

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1746

bmartel committed:
adding better 404 handling

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13716135763)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13716135763

bmartel committed:
only automatically handle 404 errors if not inline

bmartel committed:
linting

bmartel committed:
fix errorFilter checking resulting in implementation breakages with empty result error key

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13783174190)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1746'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13783174190

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13783174190
